From 5b860ab14cf99efe6fddba28091ea05e536ebf33 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Tue, 18 Feb 2020 14:42:04 +0200
Subject: [PATCH] [SW-8229] goya2: fix spi controller masking

Mask wasn't setting spi_en variable correct, set it
specificly to 1 if mask operation isn't equal to zero

Change-Id: Ie4ec283da5b143aab20078eb6eee4637f2d7d0bd
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/pcie.c b/common/pcie.c
index ded39a4..25a32e7 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -84,8 +84,10 @@
 
 void pcie_recovery(void)
 {
-	uint8_t spi_en = arcboot_rom_global.config_fuse &
-			EFUSE_SYSCFG_SPI_CONTROLLER;
+	uint8_t spi_en = 0;
+
+	if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SPI_CONTROLLER) != 0)
+		spi_en = 1;
 
 	arcboot_rom_status_update(PROGRESS_PCIE_IN_RECOVERY);
 
